src/Language/PureScript/Ide/Imports.hs

```diff
@@ -256,13 +256,14 @@ addExplicitImport' decl moduleName qualifier imports =
 ideSpan :: P.SourceSpan
 ideSpan = P.internalModuleSourceSpan "<psc-ide>"
 
+-- | If none of the elements of the list satisfy the given predicate 'predicate', then prepend the default value 'def'
+-- to the given list. Otherwise, update the first element of the list that satisfies 'predicate' with the updating
+-- function 'update'.
 updateAtFirstOrPrepend :: (a -> Bool) -> (a -> a) -> a -> [a] -> [a]
-updateAtFirstOrPrepend p t d l =
-  case findIndex p l of
-    Nothing -> d : l
-    Just i ->
-      let (x, a : y) = splitAt i l
-      in x ++ [t a] ++ y
+updateAtFirstOrPrepend predicate update def xs =
+  case break predicate xs of
```

Comment:
Another case where using `break` solves both a `-Wincomplete-uni-patterns` warning and also removes an unnecessary traversal of the list at the same time. I thought I'd take the opportunity to use clearer argument names here too.